src/driver-hidpp20.c

```diff
@@ -1398,10 +1414,14 @@ hidpp20drv_commit(struct ratbag_device *device)
 
 		ratbag_profile_for_each_resolution(profile, resolution) {
 			if (resolution->is_active) {
-				log_raw(device->ratbag, "dpi index: %d, profile %d\n", resolution->index, profile->index);
+				log_debug(device->ratbag, "Setting resolution %d to active (hidpp20)\n",resolution->index);
```

Comment:
Can you keep the original outputs here? I am afraid this will generate too much noise.

It is helpful when debugging issues specifically related to this but otherwise does not provide much help. When you need to debug such issues you can put the messages there yourself.

Reply:
I think a debug mode should log everything the program is trying to do, because if something fails for the user he at least can try to understand whats going wrong, the pure raw output is not helpful for the user. Also the first output is kinda random in my opinion since you can't even see that is about the active resolution index.

Reply:
I could mark them as raw though, but I don't think removing is a good idea.

Reply:
This outputs will get triggered every time the user changes any setting in Logitech device that has onboard memory (`0x8100`) for every profile. So, in a mouse with 3 profiles, 5 (resolutions) * 3 (profiles) = 15 new lines of debug output. This 15 new lines of output will be present for every single action the user performs, they are completely irrelevant for most actions.

When we ask a user for logs we always ask for `--verbose=raw`. This information is already contained there. Yes, it is a little bit harder to read but it there.

Due the nature of the project we don't expect users to try to be that proactive. If they are indeed that motivated they should be perfectly capable to put the log outputs there.

IMO this is a simple case of it's too little to be meaningful and it's too much for an overview.

Does that make sense?

Reply:
> When we ask a user for logs we always ask for `--verbose=raw`. This information is already contained there. Yes, it is a little bit harder to read but it there.

I can see your point but I still disagree. Anyway, this wouldn't apply to this output since I think it can be formulated better, but I can remove the other ones. Maybe opinions from @whot @bentiss?

Reply:
raw or debug, I don't really care TBH. Though `raw` were supposed to track all HW communication, when debug is one level above.

So I think switching to `debug` is fine. (do we even have users using `debug` for ratabgd?).

However, the new code removed the `dpi_index` reference, and we rather keep it.

Reply:
> However, the new code removed the `dpi_index` reference, and we rather keep it.

Only removed the profile index. But good point, gonna rewrite it. I wait for @whot for an opinion, then we should have a decision.

Reply:
Can you provide a sample output? I want to see how this will affect the log.

Reply:
Resolution 1 from disabled to 50 dpi:
```
ratbag debug: profile 0 resolution 1: set enabled (libratbag)
ratbag debug: profile 0 resolution 1: set 50 dpi (libratbag)
ratbag debug: profile 0 resolution 0: set active (hidpp20)
ratbag debug: profile 0 resolution 0: set 1000x1000 dpi (hidpp20)
ratbag debug: profile 0 resolution 1: set 50x50 dpi (hidpp20)
ratbag debug: profile 0 resolution 2: set disabled (hidpp20)
ratbag debug: profile 0 resolution 2: set 0x0 dpi (hidpp20)
ratbag debug: profile 0 resolution 3: set disabled (hidpp20)
ratbag debug: profile 0 resolution 3: set 0x0 dpi (hidpp20)
ratbag debug: profile 0 resolution 4: set disabled (hidpp20)
ratbag debug: profile 0 resolution 4: set 0x0 dpi (hidpp20)
```

Resolution 1 from 50 dpi to disabled:
```
ratbag debug: profile 0 resolution 1: set disabled (libratbag)
ratbag debug: profile 0 resolution 0: set active (hidpp20)
ratbag debug: profile 0 resolution 0: set 1000x1000 dpi (hidpp20)
ratbag debug: profile 0 resolution 1: set disabled (hidpp20)
ratbag debug: profile 0 resolution 1: set 0x0 dpi (hidpp20)
ratbag debug: profile 0 resolution 2: set disabled (hidpp20)
ratbag debug: profile 0 resolution 2: set 0x0 dpi (hidpp20)
ratbag debug: profile 0 resolution 3: set disabled (hidpp20)
ratbag debug: profile 0 resolution 3: set 0x0 dpi (hidpp20)
ratbag debug: profile 0 resolution 4: set disabled (hidpp20)
ratbag debug: profile 0 resolution 4: set 0x0 dpi (hidpp20)
```

Note that I have a high output because four of my resolutions are disabled (tbh the disabled output is a little bit useless with current hidpp20 devices). The output isn't short but it's only shown in debug (and raw) mode, not in the default output.

Reply:
Yes, that is my point. I am fine with you improving the original output but adding all of this seems unnecessary.

Reply:
> Yes, that is my point. I am fine with you improving the original output but adding all of this seems unnecessary.

But that's only because for current hidpp20 devices zero dpi and disabled _is_ the same. The whole point of adding this property was to differentiate between that, so I think it's a good debug output to see that everything is right. For example, right now I have some troubles with the piper implementation and the output (including disabled and dpi) really helps. If you just set one value and commit, yeah maybe it looks unnecessary, but piper sets sooo much values without commiting, I was really glad this output was there. The debug log exists to help development of drivers or DBus programs, and here it really helps, and the raw log is super long anyway that this doesn't make a difference.

Reply:
> now I have some troubles with the piper implementation and the output

I have no doubt about that. This is what we usually do when debugging things. But after you sort all that out and everything is working fine that code should keep working as long as nobody touches that code. After you finish working on this you don't need that output.

> so I think it's a good debug output to see that everything is right

I disagree. A debug output is not there to verify that "everything is right", it is there to provide diagnostics to the developers to try and figure out what could be triggering an issue. It should provide an overview. If we started printing out every single detail similar to this the log would be huge. I am not opposed to doing that, but please do it in a separate verbosity level.

> The debug log exists to help development of drivers or DBus programs

Yes, but as I said above, IMO it should only provide an overviw. Then you add your specific logs to help with the task.

I have a lot of devices with 3~6 profiles. 10 (lines) * 6 (profiles) = 60 new lines of debug output that I will almost never need.

Can we please meet halfway and join some of those lines? Also, can you use a prefix instead of suffix to indicate `hidpp20`/`libratbag`?

Othe thing, can we drop the dpi_y? No device uses that and I really doubt we will add such devices in the near future (or even the not so near future TBH).

Proposed output:
```
ratbag debug: ratbagd: profile 0 resolution 1: enabled, 50dpi
ratbag debug: hidpp20: profile 0 resolution 0: active, 1000 dpi
ratbag debug: hidpp20: profile 0 resolution 1: disabled, 50 dpi
ratbag debug: hidpp20: profile 0 resolution 2: disabled, 0 dpi
ratbag debug: hidpp20: profile 0 resolution 3: disabled, 0 dpi
ratbag debug: hidpp20: profile 0 resolution 4: disabled, 0 dpi
```

Reply:
> I have a lot of devices with 3~6 profiles. 10 (lines) * 6 (profiles) = 60 new lines of debug output that I will almost never need. Now imagine having 5 of them connected, which is common.

But why would you run ratbagd with --verbose=debug in the first place? It's not like that there is any useful information in there anyway, especially after the daemon started (at least in my opinion). And as I said, I don't see the problem with raw output. I even think it's easier if you see what you are actually writing before you get the raw outputs. It's actually the same when reading the features, just the other way around. But I think we should stop the discussion about the necessity at this point and wait for @whot's opinion.

> If you don't want to remove them then can we please meet halfway and join some of those lines? Also, can you use a prefix instead of suffix to indicate `hidpp20`/`libratbag`?

Had the same idea yesterday :D

> Other thing, can we drop the dpi_y? No device uses that and I really doubt we will add such devices in the near future (or even the not so near future TBH).

Is there really not a single device that has this feature? Like not only in libratbag, but which on can actually buy. Cause if not, we should drop the support completely tbh, I can't see why anyone would want that anyway. But regarding the output, I already changed to only output `dpi_x` if we don't have the capability.

> Proposed output:
> 
> ```
> ratbag debug: ratbagd: profile 0 resolution 1: enabled, 50dpi
> ratbag debug: hidpp20: profile 0 resolution 0: active, 1000 dpi
> ratbag debug: hidpp20: profile 0 resolution 1: disabled, 50 dpi
> ratbag debug: hidpp20: profile 0 resolution 2: disabled, 0 dpi
> ratbag debug: hidpp20: profile 0 resolution 3: disabled, 0 dpi
> ratbag debug: hidpp20: profile 0 resolution 4: disabled, 0 dpi
> ```
> 
> This way is smaller/more compact and straight to the point/less clutter.

That's actually a really good idea, I'll see what I can do.

Reply:
>> Other thing, can we drop the dpi_y? No device uses that and I really doubt we will add such devices in the near future (or even the not so near future TBH).
>>
> Is there really not a single device that has this feature? Like not only in libratbag, but which on can actually buy. Cause if not, we should drop the support completely tbh, I can't see why anyone would want that anyway. But regarding the output, I already changed to only output dpi_x if we don't have the capability.

IIRC the G500s had it (so hid++ 1.00), and the first Etekcity we had in our hands.

I am not 100% sure, but maybe Roccat has it too.

No idea who uses it, but I am sure some people believe it's best to have separate dpis because *whatever reason they could find out*.